### DISCUSSION TITLE
Are all nodes automatically updated?

### DISCUSSION THREAD

**Discussion by gefloh** [mysticaltech/terraform-hcloud-kube-hetzner#692] [Q&A] [answered]
Hello,

just for the sake of understanding: With the auto update of MicroOS enabled and using a HA setup, are there all nodes updated? Or just the control plane nodes? For my cluster it appears to be the latter only.

Cheers,
Gerhard

**Comment by mysticaltech** [comment, marked as answer]
@gefloh All nodes are updated normally! Now to do that, kured must be able to drain a node, and maybe if you are low on ressources it's not able to do that.

So first have a look at kured logs, and second try adding another agent to give more breathing room to the system.

Also, this bug in old versions of the module could also be the reason https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/issues/562#issuecomment-1412653699.

**Reply by gefloh** [reply]
It turned out that the auto-update was indeed turned off on the agents. I started with a non HA setup and then changed to HA later. Of course I changed the corresponding flag then in the kube.tf as well but it seems, this change has not been applied to the existing agents. Anyway, it's working like a charm now.
